src/ui/public/kfetch/index.js

```diff
@@ -32,7 +32,16 @@ class FetchError extends Error {
   }
 }
 
-export async function kfetch(fetchOptions, kibanaOptions) {
+export function kfetch(fetchOptions, kibanaOptions, isAbortable = false) {
+  let signal;
+  let abort;
+
+  if (isAbortable) {
+    const abortController = new AbortController();
+    signal = abortController.signal;
+    abort = abortController.abort.bind(abortController);
+  }
```

Comment:
Maybe abstract this into a separate `createAbortable` method that returns `{ signal, abort }`

Reply:
Good idea!